scripts/Tools/compare_test_results

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+
+"""
+Analyze results from a test root area, comparing non-BFB
+changes. Purpose is, instead of re-running tests
+in generate mode, which is very slow, allow for very fast analsis of diffs.
+
+You may need to load modules for cprnc to work.
+"""
+
+from standard_script_setup import *
+
+from CIME.utils import expect
+from CIME.XML.machines import Machines
+from CIME.compare_test_results import compare_test_results
+
+import argparse, sys, os, doctest
+
+_MACHINE = Machines()
+
+###############################################################################
+def parse_command_line(args, description):
+###############################################################################
+    parser = argparse.ArgumentParser(
+usage="""\n%s [-n] [-r <TESTROOT>] [-b <BRANCH>] [-c <COMPILER>] [<TEST> <TEST> ...] [--verbose]
+OR
+%s --help
+OR
+%s --test
+
+\033[1mEXAMPLES:\033[0m
+    \033[1;32m# From most recent run, compare all changes \033[0m
+    > %s
+    \033[1;32m# From most recent run, compare only changes for test foo and bar only \033[0m
+    > %s -n foo bar
+    \033[1;32m# From most recent run of jenkins, compare history changes for next \033[0m
+    > %s -r /home/jenkins/acme/scratch/jenkins -b next
+""" % ((os.path.basename(args[0]), ) * 6),
+
+description=description,
+
+formatter_class=argparse.ArgumentDefaultsHelpFormatter
+)
+
+    default_baseline_name = CIME.utils.get_current_branch(repo=CIME.utils.get_cime_root())
+    default_compiler      = _MACHINE.get_default_compiler()
+    scratch_root          = _MACHINE.get_value("CESMSCRATCHROOT")
+    default_testroot      = os.path.join(scratch_root)
+
+    CIME.utils.setup_standard_logging_options(parser)
+
+    parser.add_argument("-b", "--baseline-name", default=default_baseline_name,
+                        help="Name of baselines to use, corresponds to branch used.")
+
```

Comment:
Also need a way to specify baseline root: At least in my workflow, I often used baseline_gen_comp with a baseline_root that differed from the machine default (because I don't want to clutter up the standard baseline area with all of the one-off baselines that I generate).